terminus-ui/src/form-field/form-field.component.scss

```diff
@@ -0,0 +1,368 @@
+// Mocks show half of the text size, but this margin is applied to an element with the subscript
+// text font size, so we need to divide by the scale factor to make it half of the original text
+// size.
+$subscript-margin-top: .5em / $subscript-font-scale;
+// The padding applied to the form-field-wrapper to reserve space for the subscript, since it's
+// absolutely positioned. This is a combination of the subscript's margin and line-height, but we
+// need to multiply by the subscript font scale factor since the wrapper has a larger font size.
+$wrapper-padding-bottom: ($subscript-margin-top + $line-height) * $subscript-font-scale;
+// The amount we offset the label from the input text in the outline appearance.
+$outline-appearance-label-offset: -.25em;
+
+
+
+//
+// @component
+//  Form Field
+// @description
+//  The wrapping component around form controls
+//
+.ts-form-field {
+  @include reset;
+  display: block;
+
+  /*
+   *.ts-select[readonly] & {
+   *  cursor: cursor(not-allowed);
+   *}
+   */
+
+  /////////////////////////
+  // Dynamic Classes
+  /////////////////////////
+  &.ts-form-field--focused,
+  &.ts-form-field--invalid {
+    .ts-form-field__outline {
+      opacity: 0;
+      transition: opacity 100ms $swift-ease-out-timing-function;
+    }
+
+    .ts-form-field__outline--thick {
+      opacity: 1;
+    }
+  }
+
+  &:not(.ts-form-field--disabled) .ts-form-field__container:hover {
+    &.ts-form-field__outline {
+      opacity: 0;
+      transition: opacity 600ms $swift-ease-out-timing-function;
+    }
+
+    &.ts-form-field__outline--thick {
+      opacity: 1;
+    }
+  }
+
+  &.ts-form-field--float {
+    .ts-form-field__label {
+      transform:
+        translateY(-$infix-margin-top - ($infix-padding + $outline-appearance-label-offset))
+        scale($subscript-font-scale)
+      ;
+      width: 100% / $subscript-font-scale;
+    }
+
+    .ts-form-field__outline-gap {
+      border-top-color: transparent;
+    }
+  }
+
+  &.ts-form-field--focused {
+    .ts-form-field__outline--thick {
+      color: $outline-color-primary;
+    }
+
+    &.ts-form-field--accent .ts-form-field__outline--thick {
+      color: $outline-color-accent;
+    }
+
+    &.ts-form-field--warn .ts-form-field__outline--thick {
+      color: $outline-color-warn;
+    }
+  }
+
+  // Class repeated so that rule is specific enough to override focused accent color case.
+  &.ts-form-field--invalid.ts-form-field--invalid {
+    .ts-form-field__outline--thick {
+      color: $outline-color-warn;
+    }
+  }
+
+  &.ts-form-field--disabled {
+    .ts-form-field__label {
+      color: $label-disabled-color;
+    }
+
+    .ts-form-field__outline {
+      color: $outline-color-disabled;
+    }
+
+    .ts-form-field__container {
+      cursor: cursor(not-allowed);
+    }
+  }
+
+
+
+
+  /////////////////////////
+  // START: Outline
+  /////////////////////////
+
+  // <div> wrapper for outline border divs
```

Comment:
comment

Reply:
This file is so large, the comments were added to help navigate around. We could potentially split the file into imports if that feels better?